### PR TITLE
ORC-629: PPD: Floating point NaN is not transitive across comparisons

### PR DESCRIPTION
**Pull request by pgaref**

### CODE REVIEW

File: java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java [hunk 1]
```diff
@@ -495,6 +495,14 @@ static TruthValue evaluatePredicateProto(OrcProto.ColumnStatistics statsProto,
                    " include ORC-517. Writer version: {}",
           predicate.getColumnName(), writerVersion);
       return TruthValue.YES_NO_NULL;
+    } else if (category == TypeDescription.Category.DOUBLE) {
+      DoubleColumnStatistics dstas = (DoubleColumnStatistics) cs;
+      if (!Double.isFinite(dstas.getMinimum()) || !Double.isFinite(dstas.getMaximum())
```
Comment (omalley) [inline java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java:500]: This check is overly broad. Positive and Negative infinity work fine with min and max. You should just check for NaN in the sum.

Comment (wgtmac) [reply java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java:500]: I found the [fix for C++](https://github.com/apache/orc/commit/f6554377828b66d1198fe720ae8b9a734255a2df
) has been merged already. Need to ensure they are consistent.

Comment (pgaref) [reply java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java:500]: Hey @wgtmac -- the logic is replicated across Cpp and Java versions.

Regarding the broad check, the way we compare Double Stats (without Double.compare) can also lead to wrong min/max values (see test case) -- this is also something we could improve. https://github.com/apache/orc/blob/ca33ce64bf1fa8b3696e2e44b32237d842c70df3/java/core/src/java/org/apache/orc/impl/ColumnStatisticsImpl.java#L563

However, we could probably replace the infinity check with a Nan check.

Comment (wgtmac) [reply java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java:500]: Ok, I found the fix for java is merged too: https://github.com/apache/orc/commit/5dd3b29a2f4a97aa716b301c657c1853cc27e6c2. Would like to close this PR to make it less confusing.

**Pull request closed without merging**